js-core-1/week-2/lesson.md

```diff
@@ -237,262 +294,210 @@ function satisfiesRequirements(num) {
+let mentor1 = "Daniel";
+let mentor2 = "Irina";
+let mentor3 = "Rares";
 ```
 
 ...then it's probably time to use an **array**!
 
 Arrays are data structures that hold a list of values.
 
 ```js
-var mentors = ["Daniel", "Irina", "Rares"];
+let mentors = ["Daniel", "Irina", "Rares"];
 ```
 
 Arrays can hold any type of value (although almost always you only have one data type per array).
 
 ```js
-var testScores = [16, 49, 85];
-var grades = ["F", "D", "A"];
-var greetings = ["Hello, how are you?", "Hi! Nice to meet you!"];
-```
-
-## Array properties
-
-Arrays, like strings, have a `length` property.
-
-You can check this by starting a node console in your terminal.
-
-```sh
-$ node
-> var arr = [1, 2, 3];
-undefined
-> arr
-[1, 2, 3]
-> arr.length
-3
-```
-
-## Array Getters and Setters
-
-You can **get** a single value out of an array using **bracket notation**.
-
-```sh
-$ node
-> var ingredients = ["Flour", "Water", "Salt"];
-undefined
-> ingredients[0]
-Flour
-> ingredients[1]
-Water
-> ingredients.length
-3
+let testScores = [16, 49, 85];
+let grades = ["F", "D", "A"];
+let greetings = ["Hello, how are you?", "Hi! Nice to meet you!"];
 ```
 
-Did you notice how we use `[0]` to get the first value? In programming we count starting at zero.
-
-> The number inside of the brackets is called an **index**. Index just means the position of the item within the array.
-
-You can also **set** a value using bracket notation and an assignment operator (`=`).
+You can access elements in an array using the **index** of an element with **bracket notation** (remember: all arrays start at position 0! So to access the first element in an array, you need to access index 0!)
 
 ```js
-var scores = [80, 41, 47];
+let students = ["Ahmed", "Maria", "Atanas", "Nahidul", "Jack"]
 
-scores[2] = 29; // Change the last score
-scores[3] = 51; // Add a new score
+students[0] // Ahmed
+students[3] // Nahidul
 ```
 
-## Array methods
-
-Do you remember how strings have special functions called methods? Don't worry if not! Here's an example to jog your memory:
-
-```sh
-$ node
-> var name = "Daniel"
-undefined
-> name.toLowerCase()
-daniel
-```
-
-Arrays also have several methods that you can use.
-
-### `.sort()`
-
-_An array method that sorts the values in an array into ascending alphabetical or numerical order._
+You can also _assign_ new values to parts of an array:
 
 ```js
-var unorderedLetters = ["z", "v", "b", "f", "g"];
-var orderedLetters = unorderedLetters.sort();
-
-var unorderedNumbers = [8, 5, 1, 4, 2];
-var orderedNumbers = unorderedNumbers.sort();
+let students = ["Ahmed", "Maria", "Atanas", "Nahidul", "Jack"]
 
-console.log(orderedLetters); // logs [ 'b', 'f', 'g', 'v', 'z' ]
-console.log(unorderedLetters); // logs [ 'b', 'f', 'g', 'v', 'z' ]
+students[2] = "Bianca"
 
-console.log(orderedNumbers); // logs [ 1, 2, 4, 5, 8 ]
-console.log(unorderedNumbers); // logs [ 1, 2, 4, 5, 8 ]
+console.log(students) // ["Ahmed", "Maria", "Bianca", "Nahidul", "Jack"]
```

Comment:
It's good we bring this up, but as a bikeshedding comment, I would like to chat about when we teach good practise and move on (in this case not mutating arrays) - but potentially brining it up as a thing they might see, but shouldn 't do.

Going into an interview knowing that mutation can cause bugs and create issues would be a definite plus to someone going into an interview.

Reply:
Not sure who to ping on this ☝️

Reply:
Hmm, good question. I think say to play it by ear. If you feel confident about the students, your explanation and have enough time, then go for it. One thing to keep in mind is that you have the experience of the "real-world", and it's useful to talk about it, if the time is right.

I'd suggest trying to ground it in an analogy like "changing the contents" of the array, but name-checking mutation so that they know there is some jargon here. I think it's also important to discuss _why_ it's a best practice: something along the lines of if you mutate it, then you can't "trust" whats inside it?

Finally I'd be careful with words like "shouldn't", as students can interpret that very literally. I'd go with some wording like "avoid" or "most of the time it's a bad idea" (as It Depends™️).